lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -0,0 +1,184 @@
+from abc import abstractmethod
+from PIL import Image
+from typing import Any, Union, List, Dict
+from sycamore.data import BoundingBox, Element
+from sycamore.utils.cache import DiskCache
+from pathlib import Path
+from io import IOBase, BytesIO
+from sycamore.utils.pdf import pdf_to_image_files
+from sycamore.utils.import_utils import requires_modules
+from sycamore.transforms.text_extraction.text_extractor import TextExtractor
+import logging
+from sycamore.utils.time_trace import LogTime, timetrace
+import tempfile
+import os
+
+ocr_cache = DiskCache(str(Path.home() / ".sycamore/OCRCache"))
+
+logger = logging.getLogger(__name__)
+
+
+class OCRModel(TextExtractor):
+
+    @abstractmethod
+    def get_text(self, image: Image.Image) -> str:
+        pass
+
+    @abstractmethod
+    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+        pass
+
+    @timetrace("OCR Extraction")
+    def extract(self, filename: Union[str, IOBase], hash_key: str, use_cache=False, **kwargs) -> List[List[Element]]:
+        cached_result = ocr_cache.get(hash_key) if use_cache else None
+        if cached_result:
+            logger.info(f"Cache Hit for OCR. Cache hit-rate is {ocr_cache.get_hit_rate()}")
+            return cached_result
+        else:
+            with tempfile.TemporaryDirectory() as tempdirname:  # type: ignore
+                if isinstance(filename, IOBase):
+                    temp_file = tempfile.NamedTemporaryFile(prefix="ocr-pdf-input-", delete=False)
+                    with LogTime("write_pdf"):
+                        filename.seek(0)
+                        data = filename.read()
+                        temp_file.write(data)
+                        del data
+                        temp_file.flush()
+                    temp_file.close()
+                    file_name = temp_file.name
+                else:
+                    file_name = filename
+                pages = []
+                for path in pdf_to_image_files(file_name, Path(tempdirname)):
```

Comment:
I see. So this is still doing double conversion? Is it possible to do the pdf 2 image conversion once, and then share that across both the detr and text_extraction processes?